### PR TITLE
fix(curriculum) replace single-line blocks with multi-line blocks for…

### PR DESCRIPTION
**Pull request by LaurentLabine**
Data visualization and Coding Interview Prep portions.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Checks Data Visualization and Coding Interview prep on #41518

### CODE REVIEW

**Comment by LaurentLabine** [comment]
@nhcarrigan I wound up in a weird situation here where my changes of the previous PR were being added to the changes when I tried to submit the new ones because they weren't integrated into the product as of yet so I made a new Branch in order to isolate the minimum changes.  How would you have gone around it?

Hopefully, I'll manage to redeem myself with this one.

Thanks

**Comment by nhcarrigan** [comment]
It sounds like you created this new branch off of your `front-end-code-blocks` branch.

In general, when I'm starting a new change to the code-base (for a new PR), I do a `git checkout main` to switch back to the default branch. Then I make sure my branch is in-sync with the base freeCodeCamp/freeCodeCamp repository with:
```bash
git fetch upstream
git reset --hard upstream/main
git push -f
```
Note that you'll need to configure the upstream remote before this will work.

Then I create my new branch from the `main`, to ensure I'm working off a clean slate.

I'll give this a review momentarily, but if you haven't done so already I encourage you to join our [contributors chat room](https://chat.freecodecamp.org/channel/contributors) as it is a great place to ask questions regarding contributing to our codebase (plus we have additional chatrooms that are also helpful, and a ton of friendly folks ready to assist). 🙂

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/execute-a-markov-algorithm.md [hunk 1]
```diff
@@ -42,11 +42,15 @@ a never used -> .terminating rule
 
 Sample text of:
 
-`I bought a B of As from T S.`
+```html
+I bought a B of As from T S.
+```
```
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/execute-a-markov-algorithm.md:47]: @RandellDawson what are your thoughts on this file?

These aren't HTML, but technically JavaScript - though they aren't formatted as strings so would likely throw an error. Would it be better to format them as strings and use `js` as the language, or remove the codeblocks and use `pre` tags (as we do in the ruleset just above this). 

I'm leaning more toward the `pre` tags myself.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
I'm leaving a note on one of the challenges that I think needs discussion.

Aside from that, the issue indicates there are nine data visualisation challenges which need corrected, but your PR appears to only fix two of them.

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/execute-a-markov-algorithm.md (hunk 1 above)
Comment (RandellDawson) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/execute-a-markov-algorithm.md:47]: Or we can make them inline code blocks of the previous sentence like we did in the other PR.

Comment (LaurentLabine) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/execute-a-markov-algorithm.md:47]: So, Was that a consensus on this?  I could make the changes and move forward maybe implementing the last portion of it as well

Comment (RandellDawson) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/execute-a-markov-algorithm.md:47]: @LaurentLabine The consensus is to make this particular code block an inline code block a part of a sentence.  See my suggestion for the first **Ruleset** that could be applied to the other 4 rulesets.

**Change:**

````md
Sample text of:

```html
I bought a B of As from T S.
```

Should generate the output:

```html
I bought a bag of apples from my brother.
```

````

**to the following:**

````md
Sample text of `I bought a B of As from T S.` should generate the output `I bought a bag of apples from my brother.`
````

**Comment by LaurentLabine** [comment]
Made the change and it should be ready to go.

Thanks!

**Review by gikf — DISMISSED** [review]
There seem to be missing corrections in files from the `data-visualization-with-d3` sub-section of the data visualization.

**Comment by LaurentLabine** [comment]
You are right!  Should be fixed now!

**Comment by gikf** [comment]
One more file as it seems - https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/add-classes-with-d3.md

**Comment by LaurentLabine** [comment]
You are right again. Changes uploaded

**Review by RandellDawson — APPROVED** [review]
LGTM

**Comment by RandellDawson** [comment]
@nhcarrigan I think this one is good to go now.

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me. Thanks for all the hard work on this one.

**Comment by nhcarrigan** [comment]
Thank you for your continued contribution to our project. We have reviewed your pull request and are happy to accept these changes.

Please continue to watch for issues labelled `help wanted`, as these will be additional opportunities to contribute.

**Pull request merged**